### ISSUE TITLE
Support for ANSI

### ISSUE THREAD

**Issue by Taimoor-Tariq** [shikijs/shiki#315]
Is there a way to add support for ANSI formatting?

**Comment by blake-mealey** [comment]
I'm interested in contributing to this. Are the maintainers open to this feature?

**Comment by octref** [comment]
@blake-mealey You can create a renderer that renders to ANSI sequence. Currently we have renderers that output HTML/SVG.

**Comment by blake-mealey** [comment]
I would like to have a new language interpreter with the name `ansi` which parses ANSI codes and renders to similar html to regular code blocks.

**Comment by octref** [comment]
Do you mean a language that highlights something like `"\u001b[31mHelloWorld"`? Sure then you can contribute a language: https://github.com/shikijs/shiki/blob/main/docs/languages.md

**Comment by blake-mealey** [comment]
Yes, although I'd also want it to remove the control characters, which I'm guessing a standard grammar wouldn't be able to do, so this would need to be something more custom I'd imagine.

**Comment by Gerrit0** [comment]
If you're wanting to evaluate control codes, that sounds more like https://www.npmjs.com/package/ansi-to-html than something for a syntax highlighter...

**Comment by octref** [comment]
@blake-mealey, see the comment in the PR
